src/v/redpanda/admin_server.cc

```diff
@@ -767,27 +775,61 @@ void admin_server::register_partition_routes() {
           p.topic = ntp.tp.topic;
           p.partition_id = ntp.tp.partition;
 
-          auto assignment
-            = _controller->get_topics_state().local().get_partition_assignment(
-              ntp);
-
-          if (assignment) {
-              for (auto& r : assignment->replicas) {
+          // Logic for fetching replicas+status is different for normal
+          // topics vs. the special controller topic.
+          if (is_controller) {
+              // Controller topic is on all nodes.  Report all nodes,
+              // with the leader first.
+              auto leader_opt
+                = _metadata_cache.local().get_controller_leader_id();
+              if (leader_opt.has_value()) {
                   ss::httpd::partition_json::assignment a;
-                  a.node_id = r.node_id;
-                  a.core = r.shard;
+                  a.node_id = leader_opt.value();
+                  a.core = 0;
```

Comment:
nit: i think the core here is also a constexpr somewhere like cluster::controller::shard_id or something

Reply:
Fixed.